### PR TITLE
PSFR-1869 No user input validation on query string parameters in midd…

### PR DESCRIPTION
**Pull request by abdulrkalam**
…leware

### CODE REVIEW

File: server/routes/request-schema.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { check } from 'express-validator'
+
+const regExp = /^[A-Za-z0-9]*$/
+const schema = [
+  check('prisonerNumber').not().isEmpty({ ignore_whitespace: true }),
+  check('prisonerNumber').isLength({ min: 1 }),
+  check('prisonerNumber').not().isNumeric(),
```
Comment (mkerry-moj) [inline server/routes/request-schema.ts:7]: I think we should replace 
```
  check('prisonerNumber').not().isNumeric(),
  check('prisonerNumber').not().isAlpha(),
  check('prisonerNumber').custom(value => {
    return regExp.test(value.trim())
  }),
  ```
  With just `check('prisonerNumber').isAlphanumeric(),`
  
  we're just trying to check that no one is inserting any paths like `123/documents` or something, we don't really need to check that it looks like a real nomis ID (and prisoner number formats could change in the future)

File: server/routes/request-schema.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { check } from 'express-validator'
+
+const regExp = /^[A-Za-z0-9]*$/
+const schema = [
```
Comment (mkerry-moj) [inline server/routes/request-schema.ts:4]: Can we change the name of this to something like `prisonerNumberSchema` so it's more clear what it is for? (and export as that name too)

File: server/routes/request-schema.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  check('prisonerNumber').not().isEmpty({ ignore_whitespace: true }),
+  check('prisonerNumber').isLength({ min: 1 }),
```
Comment (mkerry-moj) [inline server/routes/request-schema.ts:6]: I think it's already definitely length > 1 if it is not empty

File: server/routes/request-schema.ts (hunk 1 above)
Comment (abdulrkalam) [reply server/routes/request-schema.ts:7]: I tried this earlier, when we use isAlphanumeric, it still hit the API when only characters or only number given. so  i go for separate function

File: server/routes/request-schema.ts (hunk 3 above)
Comment (abdulrkalam) [reply server/routes/request-schema.ts]: removed the unwanted check!

**Review by mkerry-moj — APPROVED** [review]

**Pull request merged**